// @opentiny/vue-option-group@3.16.0/lib/pc.js
import { renderless, api } from '@opentiny/vue-renderless/option-group/vue';
import { defineComponent, $prefix, props, setup } from '@opentiny/vue-common';
import '@opentiny/vue-theme/option-group/index.css';
import { withDirectives, openBlock, createElementBlock, createElementVNode, toDisplayString, renderSlot, vShow } from 'vue';

function _createForOfIteratorHelperLoose(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray(o, minLen);
}
function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "OptionGroup",
  componentName: "OptionGroup",
  props: [].concat(props, ["label", "disabled"]),
  setup: function setup$1(props2, context) {
    return setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-option-group tiny-select-group__wrap"
};
var _hoisted_2 = {
  class: "tiny-option-group__title tiny-select-group__title"
};
var _hoisted_3 = {
  class: "tiny-option-group__list tiny-select-group",
  ref: "selectGroup"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return withDirectives((openBlock(), createElementBlock(
    "ul",
    _hoisted_1,
    [createElementVNode(
      "li",
      _hoisted_2,
      toDisplayString(_ctx.label),
      1
      /* TEXT */
    ), createElementVNode("li", null, [createElementVNode(
      "ul",
      _hoisted_3,
      [renderSlot(_ctx.$slots, "default")],
      512
      /* NEED_PATCH */
    )])],
    512
    /* NEED_PATCH */
  )), [[vShow, _ctx.state.visible]]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { pc as default };
